### ISSUE TITLE
macOS + lld: false positive output from the linker: ld64.lld: <some path in a shared library>.o) has version 26.0.0, which is newer than target minimum of 11.0.0

### ISSUE THREAD

**Issue by keltia** [rust-lang/rust#159227]
### Code

```Rust
Any code that is bindings to a C/C++/ObjC library on macOS, when linking with `lld` at least.
```

### Current output

```Shell
warning: linker stderr: ld64.lld: /Users/roberto/Src/Rust/src/fetiche-ch/target/debug/deps/libblake3-0113f41d938ccb07.rlib(a1edd97dd51cd48d-blake3_neon.o) has version 26.0.0, which is newer than target minimum of 11.0.0
...
[repeat lots of these]
```

### Desired output

```Shell
This should be masked by rustc, like many other messages already.
```

### Rationale and extra context

Taken from https://blog.rust-lang.org/2026/07/09/Rust-1.97.0/
```text
Note that currently, linker_messages is a special lint that is not affected by the warnings lint group. This is intentional as rustc generally doesn't control linker output as precisely, and it's not uncommon for output to only appear on some platforms. If you are seeing what you think is a false positive output from the linker, please [file an issue](https://github.com/rust-lang/rust/issues/new/choose).
```

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.97.0 (2d8144b78 2026-07-07)
binary: rustc
commit-hash: 2d8144b7880597b6e6d3dfd63a9a9efae3f533d3
commit-date: 2026-07-07
host: aarch64-apple-darwin
release: 1.97.0
LLVM version: 22.1.6
```

**Comment by r3v5** [comment]
@rustbot claim

Hey everyone! I would like to work on this issue. I was able to reproduce it on macOS aarch64 (M-series) with Rust 1.97.1 and `rust-lld`. Minimal repro: any crate linking native code, e.g. `blake3 = "1"` with a simple `blake3::hash(b"hello")` call.

Built with `RUSTFLAGS="-C linker=rust-lld" cargo build`, produces:
```
warning: linker stderr: rust-lld: /private/tmp/lld-repro/target/debug/deps/libblake3-ccee12d8e7b50ff5.rlib(a1edd97dd51cd48d-blake3_neon.o) has version 26.4.1, which is newer than target minimum of 11.0.0
  |
  = note: `#[warn(linker_messages)]` on by default

warning: `lld-repro` (bin "lld-repro") generated 1 warning
```

**Comment by r3v5** [comment]
Hi, @keltia ! I have submitted PR with the fix. Feel free to review!

**Comment by keltia** [comment]
Hello and thanks!  Unfortunately your PR seems to have been closed by the bot…

**Comment by theemathas** [comment]
@keltia The PR in question (https://github.com/rust-lang/rust/pull/159453) was closed by oli-obk, which is a human moderator, not a bot.

**Comment by keltia** [comment]
I saw how it works, thanks.  Too bad the author used an LLM for this :(

**Comment by theemathas** [comment]
Oh, it seems like there's work (on zulip) by r3v5 on getting the PR reopened again 👀

**Comment by jyn514** [comment]
cc https://github.com/rust-lang/rust/issues/136113
